python/lsst/obs/decam/rawFormatter.py

```diff
@@ -32,8 +34,43 @@
 
 
 class DarkEnergyCameraRawFormatter(FitsRawFormatterBase):
-    translatorClass = DecamTranslator
+    translatorClass = astro_metadata_translator.DecamTranslator
     filterDefinitions = DarkEnergyCamera.filterDefinitions
 
     def getDetector(self, id):
         return DarkEnergyCamera().getCamera()[id]
+
+    def _determineHDU(self, detectorId):
+        """Determine the correct HDU number for a given detector id.
+
+        Returns
+        -------
+        fitsData: `lsst.afw.fits.Fits`
+            An read-only struct for accessing the required HDU.
+        metadata: `lsst.daf.base.PropertyList`
+            The metadata read from the header for that detector id.
+
+        Raises
+        ------
+        RuntimeError
+            Raised if detectorId is not found in any of the file HDUs
+        """
+        filename = self.fileDescriptor.location.path
+        fitsData = lsst.afw.fits.Fits(filename, 'r')
+        # NOTE: The primary header (HDU=0) does not contain detector data.
+        for i in range(1, fitsData.countHdus()):
```

Comment:
@rgruendl was pondering about this and thought that the vast majority had a fixed mapping from detector number to HDU and it might be more efficient to always try the fixed mapping first, and only scan through all the headers if the detector mapping is not right.